File: app/src/main/java/com/anthonymandra/image/TiffConfiguration.java
package com.anthonymandra.image;

public class TiffConfiguration extends ImageConfiguration {
    private boolean compress;

    @Override
    public ImageType getType() {
        return ImageType.tiff;
    }

    @Override
    public String getExtension() {
        return "tiff";
    }

    @Override
    protected void parse(String config) {
        compress = Boolean.parseBoolean(config);
    }

    @Override
    protected String convertToPreference() {
        return String.valueOf(compress);
    }

    public boolean getCompress() {
        return compress;
    }

    public void setCompress(boolean compress) {
        this.compress = compress;
    }
}

File: app/src/main/java/com/anthonymandra/image/TiffConfiguration.kt
package com.anthonymandra.image

class TiffConfiguration(val compress: Boolean = false) : ImageConfiguration() {
	override val type = ImageConfiguration.ImageType.TIFF
	override val extension ="tiff"
	override val parameters = compress.toString()

	constructor(preference: String) : this(preference.toBoolean())
}
